libmysqld/libmysql.c

```diff
@@ -619,17 +619,17 @@ static int default_local_infile_init(void **ptr, const char *filename,
 
 static int default_local_infile_read(void *ptr, char *buf, uint buf_len)
 {
-  int count;
   default_local_infile_data*data = (default_local_infile_data *) ptr;
 
-  if ((count= (int) my_read(data->fd, (uchar *) buf, buf_len, MYF(0))) < 0)
+  const size_t bytes_count= my_read(data->fd, (uchar *) buf, buf_len, MYF(0));
+  if (bytes_count == MY_FILE_ERROR)
   {
     data->error_num= EE_READ; /* the errmsg for not entire file read */
     my_snprintf(data->error_msg, sizeof(data->error_msg)-1,
 		EE(EE_READ),
 		data->filename, my_errno);
   }
```

Comment:
this should be returning < 0 on error. I'm not sure casting does this, but even if it does, its not particularly clear.

Reply:
The logic is the same. It is returning -1 on line 632 after casting. 
I can add a comment there explaining though.

Reply:
Or return -1 explicitly inside the if-block.

Reply:
I do like an explicit -1.

Compilers warn on existing casts depending on [arch/compiler/version](https://godbolt.org/z/oa6onEj9e).

Reply:
approved with this change.

Reply:
I added the `return -1` inside the if block